src/app/Recordings/ArchivedRecordingsTable.tsx

```diff
@@ -320,18 +325,28 @@ export const ArchivedRecordingsTable: React.FunctionComponent<ArchivedRecordings
 
   const handleDeleteRecordings = React.useCallback(() => {
     const tasks: Observable<any>[] = [];
-    addSubscription(
-      props.target.subscribe((t) => {
-        filteredRecordings.forEach((r: ArchivedRecording) => {
-          if (checkedIndices.includes(hashCode(r.name))) {
-            context.reports.delete(r);
-            tasks.push(context.api.deleteArchivedRecording(t.connectUrl, r.name).pipe(first()));
-          }
-        });
-      })
-    );
+    if (props.directory) {
+      const directory = props.directory;
+      filteredRecordings.forEach((r: ArchivedRecording) => {
+        if (checkedIndices.includes(hashCode(r.name))) {
+          context.reports.delete(r);
+          tasks.push(context.api.deleteArchivedRecordingFromPath(directory.jvmId, r.name).pipe(first()));
+        }
+      });
+    } else {
+      addSubscription(
+        props.target.subscribe((t) => {
+          filteredRecordings.forEach((r: ArchivedRecording) => {
+            if (checkedIndices.includes(hashCode(r.name))) {
+              context.reports.delete(r);
+              tasks.push(context.api.deleteArchivedRecording(t.connectUrl, r.name).pipe(first()));
+            }
+          });
+        })
+      );
+    }
     addSubscription(forkJoin(tasks).subscribe());
```

Comment:
Should this call be after pushing tasks into the task array (i.e. inside the first addSubscriptions)?

Reply:
something like this to make sure that the task list is populated before calling `forkJoin`? Because the current order looks like the task list is done async (possibly after `forkJoin` returns)?

```typescript
addSubscription(
  props.target.subscribe((t) => {
    const tasks: Observable<any>[] = [];
    filteredRecordings.forEach((r: ArchivedRecording) => {
      if (checkedIndices.includes(hashCode(r.name))) {
        context.reports.delete(r);
        tasks.push(context.api.deleteArchivedRecording(t.connectUrl, r.name).pipe(first()));
      }
    });
    addSubscription(forkJoin(tasks).subscribe());
  })
);
```